openquakeplatform/irv/static/irv/js/irv_viewer.js

```diff
@@ -0,0 +1,2038 @@
+            } else {
+                projDefOperatorsAreStandard = true;
+                $('.alert-custom-operators').hide();
+            }
+            loadPD(sessionProjectDef);
+            $('#iri-spinner').hide();
+            $('#project-definition-svg').show();
+            processIndicators(layerAttributes, sessionProjectDef);
+        }
+    }
+}
+
+function has_custom_fields(projDef) {
+    if (typeof(projDef.operator) !== 'undefined' && namesToOperators[projDef.operator].code == 'CUSTOM') {
+        return true;
+    }
+    for (var childIdx in projDef.children) {
+        var child = projDef.children[childIdx];
+        if (has_custom_fields(child)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function getGeoServerLayers() {
+    $('#load-project-spinner').show();
+    var IRMTLayerNames = [];
+    var IRMTLayerTitle = [];
+    var url = "http://"+ window.location.hostname +":8000/geoserver/ows?service=WFS&version=1.0.0&REQUEST=GetCapabilities&SRSNAME=EPSG:4326&outputFormat=json&format_options=callback:getJson";
+    // Get layers from GeoServer and populate the layer selection menu
+
+    $.ajax({
+        url: url,
+        contentType: 'application/json',
+        success: function(xml) {
+
+
+            //convert XML to JSON
+            var xmlText = new XMLSerializer().serializeToString(xml);
+            var x2js = new X2JS();
+
+            var jsonElement = x2js.xml_str2json(xmlText);
+
+            var featureType = jsonElement.WFS_Capabilities.FeatureTypeList.FeatureType;
+
+            // Find the IRMT keywords
+            var stringsToLookFor = ['SVIR', 'IRMT_QGIS_Plugin'];
+            // Reload if the api call was incomplete
+            if (featureType.length === undefined) {
+                getGeoServerLayers();
+                return;
+            }
+
+            for (var i = 0; i < featureType.length; i++) {
+                for (var j=0; j < stringsToLookFor.length; j++) {
+                    stringToLookFor = stringsToLookFor[j];
+                    if (featureType[i].Keywords.indexOf(stringToLookFor) > -1) {
+                        // IRMTLayerNames.push(featureType[i].Title + " (" + featureType[i].Name + ")");
+                        IRMTLayerNames.push(featureType[i].Name);
+                        IRMTLayerTitle.push(featureType[i].Title);
+                        break;
+                    }
+                }
+            }
+
+            if (IRMTLayerNames.length == 0) {
+		        $('#load-project-spinner').hide();
+		        $('#ajaxErrorDialog').empty();
+		        $('#ajaxErrorDialog').append(
+		            '<p>No layer found.</p>'
+		        );
+		        $('#ajaxErrorDialog').dialog('open');
+            }
+
+            // Create AngularJS dropdown menu
+            var mapLayerList = [];
+            for (var ij = 0; ij < IRMTLayerNames.length; ij++) {
+                var Obj = {};
+                Obj.name = IRMTLayerNames[ij];
```

Comment:
Changed this.
Before was :  tempObj.name = IRMTLayerNames[ij];

Reply:
I don't like the capital letter in Obj (and actually also the name is too generic).
I suggest to rename all occurrences of `Obj` into something like `layerProperties`